client/sites-dashboard/components/sites-ellipsis-menu.tsx

```diff
@@ -107,6 +114,69 @@ const HostingConfigItem = ( { site, recordTracks }: SitesMenuItemProps ) => {
 	);
 };
 
+const ModalContent = styled.div( {
+	padding: 16,
+	width: '80vw',
+	maxWidth: '480px',
+	minHeight: '100px',
+	display: 'flex',
+	flexDirection: 'column',
+} );
+
+const modalOverlayClassName = css( {
+	// golbal-notices has z-index: 179
+	zIndex: 178,
+} );
+
+function useSafeSiteHasFeature( siteId: number, feature: string ) {
+	const dispatch = useDispatch();
+	useEffect( () => {
+		dispatch( fetchSiteFeatures( siteId ) );
+	}, [ dispatch, siteId ] );
+
+	return useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, feature );
+	} );
+}
+
+const PreviewSiteModalItem = ( { recordTracks, site }: SitesMenuItemProps ) => {
+	const { __ } = useI18n();
+	const [ isVisible, setIsVisible ] = useState( false );
+	const openModal = () => setIsVisible( true );
+	const closeModal = () => setIsVisible( false );
+
+	const onSitePreviewClick = () => {
+		recordTracks( 'calypso_sites_dashboard_site_action_preview_link_click' );
+		openModal();
+	};
+
+	const hasSitePreviewLinksFeature = useSafeSiteHasFeature(
+		site.ID,
+		WPCOM_FEATURES_SITE_PREVIEW_LINKS
+	);
+
+	if ( ! hasSitePreviewLinksFeature ) {
+		return null;
+	}
+
+	return (
+		<>
+			<MenuItemLink onClick={ onSitePreviewClick }>{ __( 'Share site for preview' ) }</MenuItemLink>
```

Comment:
We used "Share Site for Preview" as a project codename, but I'm unsure if we should keep that name. Would it be better to call the feature "preview links" and use the "Create a preview link" CTA here?

Reply:
Maybe we can merge it and then update the texts, later this week.